RoundTripper wraper support for metrics collection

0xMALVEE opened the pull request:
needed for pr: https://github.com/miniohq/eos/pull/2231
fixes issue: https://github.com/miniohq/eos/issues/2143

0xMALVEE committed:
Update client.go

klauspost commented on kms/client.go:
```diff
@@ -50,6 +50,14 @@ type Config struct {
+	// The wrapper will be inserted between the LoadBalancer and the
+	// base transport, allowing it to observe all requests with their
+	// selected endpoints.
+	WrapRoundTripper func(http.RoundTripper) http.RoundTripper
```
Usually you'd do this by providing a custom http.Transport (or even better Roundtripper) for the client to use. Usually that is a more flexible solution to this problem.

0xMALVEE committed:
Update client.go

klauspost commented on kms/client.go:
```diff
@@ -120,8 +117,8 @@ func NewClient(conf *Config) (*Client, error) {
-	if conf.WrapRoundTripper != nil {
-		rt = conf.WrapRoundTripper(transport)
+	if conf.Transport != nil {
+		rt = conf.Transport
```
I think there is an issue is that `tlsConf` is not applied here. You know, maybe just revert what I proposed - it isn't needed here.

0xMALVEE committed:
kms: use WrapRoundTripper instead of Transport field

Replace the Transport http.RoundTripper field with WrapRoundTripper func
to provide a cleaner API for wrapping the base transport with custom
middleware (metrics, tracing, etc.).

This allows the caller to add functionality without needing to handle
TLS configuration manually.

0xMALVEE commented:
closing in favor of https://github.com/minio/kms-go/pull/43
